### ISSUE TITLE
[Feature-Request] Store Non-Steam Games Infos and associated Images in the SteamApps Folder

### ISSUE THREAD

**Issue by NightHammer1000** [ValveSoftware/steam-for-linux#8575]
**The Problem**

A great Feature, especially for the SteamDeck would be the possibility to store an Additional Shortcuts.vdf together with its User-Defined Images inside the SteamApps Folder.

This would make SD Card/External Drive swapping for Users even better because it would only show the Non-Steam Games currently available on the Connected Drive.

Currently, because the Shortcuts.vdf is stored in the userdata folder, all the Shortcuts are shown. Even those to Games that are present on other SD Cards.

**Possible Way to implement it**
One possible way to implement this would be a checkbox under the Steam Library Options that would allow a user to define a Library as the default Shortcuts storage.

Even better would be the Option to allow the User to decide if they want to add "User" Shortcuts or "Library" Shortcuts when adding Non-Steam Games.

**Comment by alkazar** [comment]
Is this a duplicate of https://github.com/ValveSoftware/steam-for-linux/issues/8439 ?

**Comment by Continous** [comment]
No; though theoretically, #8439 could allow third-party apps to solve the problem themselves, this would make non-steam apps library-specific. Since Steam now supports auto-mounting libraries, this would let Steam automagically support auto-mounting non-steam applications in a similar way to Steam games.
